src/meta-srv/src/service/admin/region_migration.rs

```diff
@@ -0,0 +1,203 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::HashMap;
+use std::num::ParseIntError;
+use std::str::FromStr;
+
+use common_meta::peer::Peer;
+use common_meta::ClusterId;
+use serde::Serialize;
+use snafu::ResultExt;
+use store_api::storage::RegionId;
+use tonic::codegen::http;
+
+use super::HttpHandler;
+use crate::error::{self, Error, Result};
+
+pub trait PeerLookup: Send + Sync {
+    fn peer(&self, peer_id: u64) -> Option<Peer>;
+}
+
+/// The handler of submitting migration task.
+pub struct SubmitRegionMigrationTaskHandler {
+    // TODO(weny): waits for https://github.com/GreptimeTeam/greptimedb/pull/3014
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+struct SubmitRegionMigrationTaskParams {
```

Comment:
Perhaps we could call it: SubmitRegionMigrationTaskRequest?

Reply:
Yes, I think the `Params` is more precise. The `SubmitRegionMigrationTaskParams ` are parsed from the body of the request. IMO, a `Request` typically should contain the context of the request.

Reply:
I changed my mind. I changed it to `SubmitRegionMigrationTaskRequest `. We can have something like `Request< SubmitRegionMigrationTaskRequest>` in the future.